src/Public/New-CCMDeploymentStep.ps1

```diff
@@ -144,18 +154,20 @@ function New-CCMDeploymentStep {
         switch ($PSCmdlet.ParameterSetName) {
             'Basic' {
                 $Body = @{
-                    Name                         = "$Name"
-                    DeploymentPlanId             = "$(Get-CCMDeployment -Name $Deployment | Select-Object -ExpandProperty Id)"
-                    DeploymentStepGroups         = @(
-                        Get-CCMGroup -Group $TargetGroup | Select-Object Name, Id | ForEach-Object {
-                            [pscustomobject]@{ groupId = $_.id; groupName = $_.name }
-                        }
+                    name                           = $Name
+                    deploymentPlanId               = (Get-CCMDeployment -Name $Deployment).id
+                    deploymentStepGroups           = @(
+                        Get-CCMGroup -Group $TargetGroup | Select-Object -Property @(
+                            @{ Name = "groupId"; Expression = { $_.id } }
+                            @{ Name = "groupName"; Expression = { $_.name } }
+                        )
                     )
-                    ExecutionTimeoutInSeconds    = "$ExecutionTimeoutSeconds"
-                    RequireSuccessOnAllComputers = "$RequireSuccessOnAllComputers"
-                    failOnError                  = "$FailOnError"
-                    validExitCodes               = "$($validExitCodes -join ',')"
-                    script                       = "$($ChocoCommand.ToLower())|$($PackageName)"
+                    executionTimeoutInSeconds      = $ExecutionTimeout
+                    machineContactTimeoutInMinutes = $MachineContactTimeout
+                    requireSuccessOnAllComputers   = $RequireSuccessOnAllComputers
+                    failOnError                    = $FailOnError

```

Comment:
```suggestion
                    executionTimeoutInSeconds      = $ExecutionTimeout
                    machineContactTimeoutInMinutes = $MachineContactTimeout
                    requireSuccessOnAllComputers   = $RequireSuccessOnAllComputers.IsPresent
                    failOnError                    = $FailOnError.IsPresent
```

JSON serialization doesn't handle these well, it treats a `[switch]` parameter object as an object with the form `{ IsPresent: true }` rather than a raw boolean, make sure we take the inner property explicitly (this should be fixed below for consistency as well)